src/EditorFeatures/CSharpTest/MisplacedUsingDirectives/AbstractMisplacedUsingDirectivesCodeFixProviderTests.cs

```diff
@@ -0,0 +1,134 @@
+";
+
+            // Capture the working directory so it can be restored after the test runs.
+            var workingDirectory = Environment.CurrentDirectory;
+
+            // Create a temporary folder so that the coding conventions library can be
+            // used to read code style preferences from an .editorconfig.
+            var testDirectoryName = Path.GetRandomFileName();
+            Directory.CreateDirectory(testDirectoryName);
+            try
+            {
+                // Change the working directory to our test directory so that we 
+                // can find the .editorconfig from the analyzer to get our 
+                // code style settings.
+                Environment.CurrentDirectory = testDirectoryName;
+
+                File.WriteAllText(".editorconfig", editorConfig);
+
+                // The contents of this file are ignored, but the coding conventions 
+                // library checks for existence before .editorconfig is used.
+                File.WriteAllText("Test0.cs", string.Empty);
+
+                // Do not specify the full path to the source file since only the
+                // filename will be copied causing mismatch when verifying the fixed state.
+                var test = CreateTest(("Test0.cs", source), fixedSource);
+
+                // Set code style settings in the OptionSet so that the CodeFix can
+                // access the settings.
+                test.OptionsTransforms.Add(
+                    optionsSet => optionsSet.WithChangedOption(GenerationOptions.PlaceSystemNamespaceFirst, LanguageNames.CSharp, placeSystemNamespaceFirst)
+                        .WithChangedOption(CSharpCodeStyleOptions.PreferredUsingDirectivePlacement, usingPlacement));
+
+                // Fix the severity of expected diagnostics.
+                var fixedExpectedResults = expected.Select(
+                    result => result.WithSeverity(usingPlacement.Notification.Severity.ToDiagnosticSeverity() ?? DiagnosticSeverity.Hidden));
+                test.ExpectedDiagnostics.AddRange(fixedExpectedResults);
+
+                // Fix the severity of remaining diagnostics.
+                var fixedRemainingResults = remaining.Select(
+                    result => result.WithSeverity(usingPlacement.Notification.Severity.ToDiagnosticSeverity() ?? DiagnosticSeverity.Hidden));
+                test.FixedState.ExpectedDiagnostics.AddRange(fixedRemainingResults);
+
+                await test.RunAsync();
+            }
+            finally
+            {
+                // Clean up by resetting the working directory and deleting
+                // the temporary folder.
+                Environment.CurrentDirectory = workingDirectory;
+                try
+                {
+                    Directory.Delete(testDirectoryName, true);
+                }
+                catch { }
+            }
+        }
```

Comment:
this is *super* strange to see.  why are the 'MisplacedUsingDirectives' tests doing *anything* around manually performing IO and/or manually manipulating editorconfigs.  This seems like a very strange test layering violation.

Reply:
Removed